gravitee-apim-console-webui/src/management/api/creation/v4/models/ApiCreationStepperService.ts

```diff
@@ -13,29 +13,31 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-import { Injectable, InjectionToken, Type } from '@angular/core';
+import { Injectable, Type } from '@angular/core';
 import { ReplaySubject, Subject } from 'rxjs';
 import { cloneDeep } from 'lodash';
 
 import { ApiCreationPayload } from './ApiCreationPayload';
 
-export const API_CREATION_PAYLOAD = new InjectionToken<ApiCreationPayload>('API_CREATION_PAYLOAD');
-
 export interface ApiCreationStep {
-  number: number;
+  position: number;
   label: string;
   component: Type<unknown>;
-  payload: ApiCreationPayload;
+  patchPayload: (lastPayload: ApiCreationPayload) => ApiCreationPayload;
   type: 'primary' | 'secondary';
 }
 
-type ApiCreationStepPayload = Omit<ApiCreationStep, 'type' | 'payload'> & { payload?: ApiCreationPayload };
+type ApiCreationStepPayload = Omit<ApiCreationStep, 'type' | 'patchPayload'>;
```

Comment:
The `ApiCreationStepPayload` looks a bit confusing to me, I had a hard time understanding its use. WDYT about `ApiCreationStepDescriptor` instead ? 

And also splitting in 2 types and having an intersection type like follows would make it easier to understand I think
```
export interface ApiCreationStepDescriptor {
  position: number;
  label: string;
  component: Type<unknown>;
}

export interface ApiCreationStepPayload {
  patchPayload: (lastPayload: ApiCreationPayload) => ApiCreationPayload;
  type: 'primary' | 'secondary';
}

export type ApiCreationStep = ApiCreationStepDescriptor & ApiCreationStepPayload;
```

Reply:
Hum,  The idea of the type ApiCreationStepPayload it is really just to init the stepper with a simplified ApiCreationStep. and it's not exported only for constructor use 🤷‍♂️ 
`type` are also not require